pkg/roachprod/install/cockroach.go

```diff
@@ -405,9 +381,15 @@ func (c *SyncedCluster) Start(ctx context.Context, l *logger.Logger, startOpts S
 	}
 
 	if c.allowServiceRegistration() {
-		err := c.maybeRegisterServices(ctx, l, startOpts, c.FindOpenPorts)
-		if err != nil {
-			return err
+		// Only register services when starting a virtual cluster, or using custom
+		// ports, or for local cluster port management to avoid collisions. The
+		// lookup logic will automatically fall back to the default ports if the
```

Comment:
In the fallback case, we still have to lookup the records in the DNS server to confirm they don't exist right? And since we don't find it, we never cache the entry and have to do the DNS lookup again every time? Compared to before, we pay the registration time up front but the subsequent lookups should be cached?

Would it be worth caching a dummy record in this case to avoid this?

Reply:
> And since we don't find it, we never cache the entry

We do cache empty records, but I'll double check to make sure.

We'll still pay the cost of one lookup (to trigger the empty record entry for that DNS name in the cache) that can technically be avoided by "warming" the cache, but that might be a bit of a strange API call to get the cache to keep an empty record for the cluster. I'll think about it a bit. But the good news is, it's only one call - and only relevant to `roachtest` mostly.

Reply:
https://github.com/cockroachdb/cockroach/pull/125385 was another proposal that would have zero lookups rather than relying on negative caching at the expense of needing to decide, at cluster creation, if we're using SRV records or not.

Reply:
I'm not opposed to the above solution to disable discovery completely. But I'd like this PR in addition since it also alleviates over usage of DNS in `roachtest` for tests that don't use virtual clusters.